Sync/SyncTelemetryUtils.swift

```diff
@@ -242,13 +242,22 @@ public protocol SyncPingFailureFormattable {
 public struct SyncPing: SyncTelemetryPing {
     public private(set) var payload: JSON
 
-    static func pingFields(prefs: Prefs, why: SyncPingReason) -> Deferred<Maybe<(token: TokenServerToken, fields: [String: Any])>> {
+    static func pingFields(
+        prefs: Prefs,
+        why: SyncPingReason,
+        completion: @escaping ((Maybe<(token: TokenServerToken, fields: [String: Any])>) -> Void)
```

Comment:
If possible let's avoid using `Maybe` type as well since this relates to `Deferred` code too. We can probably use an optional instead `?`

Reply:
I thought someone needs Maybe.failureValue to check error. I agree with you that`Maybe` could be removed. In this case, failure isn't used to. So I removed it.